tests/v2/validation/registries/registry_test.go

```diff
@@ -0,0 +1,221 @@
+package registries
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+
+	corralext "github.com/rancher/rancher/tests/framework/extensions/corral"
+	ranchersetup "github.com/rancher/rancher/tests/framework/extensions/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/registries"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	corralRegistryRancherImageName = "registryrancher"
+	corralRegistryRancherName      = "rancherglobalregistry"
+	corralAuthDisabledImageName    = "registryauthdisabled"
+	corralAuthDisabledName         = "downstreamregistrydisabled"
+	corralAuthEnabledImageName     = "registryauthenabled"
+	corralAuthEnabledName          = "downstreamregistryenabled"
+)
+
+var (
+	adminPassword = os.Getenv("ADMIN_PASSWORD")
+)
+
+type RegistryTestSuite struct {
+	suite.Suite
+	session                        *session.Session
+	podListClusterAuth             *v1.SteveCollection
+	podListClusterNoAuth           *v1.SteveCollection
+	podListClusterLocal            *v1.SteveCollection
+	clusterAuthRegistryHost        string
+	clusterNoAuthRegistryHost      string
+	localClusterGlobalRegistryHost string
+}
+
+func (rt *RegistryTestSuite) TearDownSuite() {
+	ranchersetup.RancherCleanup(rt.session)
+	corralList, err := corral.ListCorral()
+	require.NoError(rt.T(), err)
+	err = corralext.CleanUpCorrals(corralList)
+	require.NoError(rt.T(), err)
```

Comment:
I totally agree with this, we don't always want to clean up, there are times we want to be able to go back to our setup and debug manually. Rancher clean up should be done outside of this test suite to maintain reusability. The auth enabled, and the corralAuthDisabledName corral packages get cleaned up with `rt.session.Cleanup() `same with he clusters you have created. This is just being redundant. From line 51-55 should be removed.

Reply:
I see and agreed, sounds like that way we'd eliminate hardcoded cleanup, so if there's rancher cleanup glue outside as the third step; corral, and clusters would be deleted since they are registered to the session cleanup?

Reply:
If the containers share the same volume then you can simply just use the new RancherCleanup extension in a bin.